src/main/java/bio/terra/service/dataset/flight/delete/DeleteDatasetStoreProjectIdStep.java

```diff
@@ -1,31 +1,47 @@
 package bio.terra.service.dataset.flight.delete;
 
 import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetBucketDao;
 import bio.terra.service.dataset.DatasetService;
 import bio.terra.service.dataset.flight.DatasetWorkingMapKeys;
 import bio.terra.stairway.FlightContext;
 import bio.terra.stairway.FlightMap;
 import bio.terra.stairway.Step;
 import bio.terra.stairway.StepResult;
 import bio.terra.stairway.exception.RetryException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 import java.util.UUID;
 
 public class DeleteDatasetStoreProjectIdStep implements Step {
   private final UUID datasetId;
   private final DatasetService datasetService;
+  private final DatasetBucketDao datasetBucketDao;
 
-  public DeleteDatasetStoreProjectIdStep(UUID datasetId, DatasetService datasetService) {
+  public DeleteDatasetStoreProjectIdStep(
+      UUID datasetId, DatasetService datasetService, DatasetBucketDao datasetBucketDao) {
     this.datasetId = datasetId;
     this.datasetService = datasetService;
+    this.datasetBucketDao = datasetBucketDao;
   }
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    FlightMap workingMap = context.getWorkingMap();
     Dataset dataset = datasetService.retrieve(datasetId);
-    UUID projectResourceId = dataset.getProjectResourceId();
+    Set<UUID> projectResourceIds = new HashSet();
+    // main dataset google project
+    projectResourceIds.add(dataset.getProjectResourceId());
 
-    FlightMap workingMap = context.getWorkingMap();
-    workingMap.put(DatasetWorkingMapKeys.PROJECT_RESOURCE_ID, projectResourceId);
+    // google projects from ingests into dataset w/ different billing profile
+    projectResourceIds.addAll(datasetBucketDao.getProjectResourceIdsForBucketPerDataset(datasetId));
+
+    List<UUID> projectResourceIdList = new ArrayList<>(projectResourceIds);
```

Comment:
There are a few ways to go about this, this is the way I picked -- 
I'm putting things into a set to account for duplicates. However, I want the result to be in a list. 

We can get duplicates from the two checks:
- 1st call gets the dataset's google project
- 2nd call gets any bucket associated with the dataset. So, if we're ingested into the dataset w/ the default billing profile, it would also be returned here.